src/main/java/com/bereda/product_api/external_api/service/ExternalApiService.java

```diff
@@ -0,0 +1,37 @@
+package com.bereda.product_api.external_api.service;
+
+import com.bereda.product_api.external_api.model.ExternalExchangeRateResponse;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.Optional;
+
+@Slf4j
+@Service
+public class ExternalApiService {
+
+    private final RestTemplate restTemplate;
+    private final String baseUrl;
+
+    public ExternalApiService(final RestTemplate restTemplate, final @Value("${apiRequest.url}") String baseUrl) {
+        this.restTemplate = restTemplate;
+        this.baseUrl = baseUrl;
+    }
+
+    public Optional<ExternalExchangeRateResponse> exchangeRateRequest(final String fromCurrency) {
+        final String baseCurrency = "PLN";
+        try {
+            final ResponseEntity<ExternalExchangeRateResponse> exchangeRateResponseEntity =
```

Comment:
it won't work as your request is wrong. You don't pass to as param. Make Base URL just http://slepy-currency-converter.herokuapp.com/api/currencies and pass all params here - search for some method to use iwht RestTemplate